### ISSUE TITLE
Content of multids file gets erased once processed on node

### ISSUE THREAD

**Issue by danielo515** [Jermolene/TiddlyWiki5#1788]
Hello,

Today I have created a multids and I placed it under the tiddlers folder of a node.js tiddlywiki. When the server have started, all the tiddlers were processed and created, so far so good. Then, notepad++ have noticed me that the file that I had opened (the multids file) were modified. When I reloaded the file, it was totally empty.

Is this the intended behavior?

Regards

**Comment by pmario** [comment]
>Is this the intended behavior?

no, since it would break every edition, that uses them. But .multids are not intended to be used as content tiddlers. The produce shadow tiddlers, for plugins, themes, languages. ... 

There is no mechanism in the core, that can "repackage" a shadow tiddler if you change it, and save it in a single .multids file. .. At leas I don't know it :)

**Comment by pmario** [comment]
What do you try to do?

**Comment by danielo515** [comment]
I wanted to create a bunch of tiddlers with an script that process a text file. Then I though that the correct thing was to put the multids file in the tiddlers folder, to make them available on on the wiki. Note that I put it on the normal tiddlers folder, not under system neither plugins one.

**Comment by Jermolene** [comment]
Hi @danielo515 the issue is that the filesystemadaptor only knows how to write .tid files, and doesn't understand how to save a tiddler by splicing it into an existing .multids file. Any tiddlers you define in a .multids file should read OK, it's only when you try to write to them that the problem occurs.

That's why the core currently only uses .multids files within plugins.

**Comment by danielo515** [comment]
Hello @Jermolene 

I can understand that, thank you very much. Then what is the most suitable format for creating a bunch of tiddlers progragmatically?

**Comment by Jermolene** [comment]
> Then what is the most suitable format for creating a bunch of tiddlers progragmatically?

.tid files would usually be the most convenient, I'd have thought.

**Comment by pmario** [comment]
Jeremy, you marked it as a bug. ... It may be possible to write changes back to a .multids file. .. But what if the user adds a tag? .multids can't handle this. ..

**Comment by Jermolene** [comment]
You're right, I shouldn't make it a bug if it can't be fixed.

**Comment by pmario** [comment]
IMO .multids format is not up for the task. It's defined to be a light weight format for translators, to have less files that need to be manually handled. 

There have been some discussions for a different multi tiddler format for manual usage. But I think this introduces a lot of complexity and the .tid format imo is perfectly fine and works out of the box.

@danielo515 what do you think?

**Comment by danielo515** [comment]
I agree with the fact that multid files should be read only. But I don't understand why they get their contents erased after parsing.  They are useful to write bunch of Tiddlers fast and light, why not allow them not only for plugins?

**Comment by Jermolene** [comment]
> But I don't understand why they get their contents erased after parsing. 

That's a bug, I'd say. I'm not sure what we should do instead, but it definitely shouldn't be deleting stuff. I've changed the title to clarify the required fix.
